internal/driver/credentials.go

```diff
@@ -0,0 +1,118 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+package driver
+
+import (
+	"fmt"
+
+	"github.com/IOTechSystems/onvif"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+)
+
+// Credentials encapsulates username, password, and AuthMode attributes.
+// Assign AuthMode to "digest" | "usernametoken" | "both" | "none"
+type Credentials struct {
+	Username string
+	Password string
+	AuthMode string
+}
+
+const (
+	AuthModeDigest        string = onvif.DigestAuth
+	AuthModeUsernameToken string = onvif.UsernameTokenAuth
+	AuthModeBoth          string = onvif.Both
+	AuthModeNone          string = onvif.NoAuth
+)
+
+const (
+	UsernameKey = "username"
+	PasswordKey = "password"
+	AuthModeKey = "mode"
+)
+
+var (
+	noAuthCredentials = Credentials{
+		AuthMode: AuthModeNone,
+	}
+)
+
+func IsAuthModeValid(mode string) bool {
+	return mode == AuthModeDigest ||
+		mode == AuthModeUsernameToken ||
+		mode == AuthModeBoth ||
+		mode == AuthModeNone
+}
+
+// tryGetCredentials will attempt one time to get the credentials located at secretPath from
+// secret provider and return them, otherwise return an error.
+func tryGetCredentials(secretPath string) (Credentials, errors.EdgeX) {
+	secretData, err := sdk.RunningService().SecretProvider.GetSecret(secretPath, UsernameKey, PasswordKey, AuthModeKey)
+	if err != nil {
+		return Credentials{}, errors.NewCommonEdgeXWrapper(err)
+	}
+
+	credentials := Credentials{
+		Username: secretData[UsernameKey],
+		Password: secretData[PasswordKey],
+		AuthMode: secretData[AuthModeKey],
+	}
+
+	if !IsAuthModeValid(secretData[AuthModeKey]) {
+		sdk.RunningService().LoggingClient.Warnf("AuthMode is set to an invalid value: %s. setting value to 'none'.", credentials.AuthMode)
+		credentials.AuthMode = AuthModeNone
```

Comment:
Maybe we should default it to `usernametoken` or something sensible? If the creds are valid, but the mode isn't, I think we should still try and use them.